src/main/kotlin/io/openfuture/state/watcher/ethereum/EthereumWatcher.kt

```diff
@@ -0,0 +1,16 @@
+package io.openfuture.state.watcher.ethereum
+
+import kotlinx.coroutines.runBlocking
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+
+@Component
+@ConditionalOnProperty(name = ["watcher.ethereum.enabled"])
+class EthereumWatcher(private val ethereumBlockProcessor: EthereumBlockProcessor) {
```

Comment:
ethereumBlockProcessor & EthereumWatcher are the same instances. You should not have separate classes for them. At least for now.